src/typeahead/typeahead.ts

```diff
@@ -118,44 +125,45 @@ export class NgbTypeahead implements OnInit,
   ngOnDestroy() { this._subscription.unsubscribe(); }
 
   ngOnInit() {
-    this._subscription = this._valueChanges.let (this.ngbTypeahead).subscribe((results) => {
-      if (!results || results.length === 0) {
-        this.closePopup();
-      } else {
-        this._openPopup();
-        this._windowRef.instance.results = results;
-        this._windowRef.instance.term = this._elementRef.nativeElement.value;
-        if (this.resultFormatter) {
-          this._windowRef.instance.formatter = this.resultFormatter;
-        }
-        if (this.resultTemplate) {
-          this._windowRef.instance.resultTemplate = this.resultTemplate;
-        }
-      }
-    });
+    this._subscription =
+        this._valueChanges.do(value => this._userInput = value).let (this.ngbTypeahead).subscribe((results) => {
```

Comment:
This looks strange - we should probably not be putting a space after `let` there. Ideally, I think we should probably shorten this line by dropping to the next, but we might be dealing with clang-format here...